File: 실습/6장/06_partition/Main.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        List<Integer> numbers =
                Arrays.asList(1, 2, 3, 4, 5);

        List<Integer> evenNumbers = new ArrayList<>();
        List<Integer> oddNumbers = new ArrayList<>();

        for (int number : numbers) {
            if (number % 2 == 0) {
                evenNumbers.add(number);
            } else {
                oddNumbers.add(number);
            }
        }

        System.out.println("짝수: " + evenNumbers);
        System.out.println("홀수: " + oddNumbers);
    }
}

File: 실습/6장/06_partition/Main.kt
fun main() {
    val numbers = listOf(1, 2, 3, 4, 5)

    val (evenNumbers, oddNumbers) = numbers.partition {
        it % 2 == 0
    }

    println("짝수: $evenNumbers")
    println("홀수: $oddNumbers")
}